modules/heat_conduction/src/kernels/HeatConductionTimeDerivative.C

```diff
@@ -17,7 +17,7 @@ InputParameters validParams<HeatConductionTimeDerivative>()
   // over current volume by setting the use_displaced_mesh flag.
   params.set<bool>("use_displaced_mesh") = true;
 
-  params.addParam<MaterialPropertyName>("specific_heat", "specific_heat",
+  params.addParam<MaterialPropertyName>("specific_heat_name", "specific_heat",
```

Comment:
Well then we need to fix `density_name` in the *same* file which is the other way around.